AK/StdLibExtras.h

```diff
@@ -452,6 +452,14 @@ template<typename Base, typename Derived>
 struct IsBaseOf : public IntegralConstant<bool, __is_base_of(Base, Derived)> {
 };
 
+template<typename T>
+struct IsTrivial : public IntegralConstant<bool, __is_trivial(T)> {
+};
+
+template<typename T>
+struct IsTriviallyCopyable : public IntegralConstant<bool, __is_trivially_copyable(T)> {
+};
```

Comment:
Don't mind me if you're just doing what STL does, but does making this a type provide any advantages over a function?

```cpp
template<typename T>
constexpr bool is_trivial()
{
    return __is_trivial(T);
}
```

Reply:
I don't know if there are any advantages, other than maybe being able to work on older c++ standards? I just did what we do for the other ones, which happens to be what the stl does.

Reply:
I don't think we need to or want to support older standards, or do we? Perhaps in AK we have to, because of the host...

cc @awesomekling

Reply:
Well, as someone being more accustomed to the stl, this feels more natural and correct. The `constexpr` function is_trivial looks like a regular function, and I probably would never guess that it's meant to be used in `static_assert`. But that's just me. In your defense, c++17 adds the `_v` variants, which is essentially the same.

Reply:
I don't think we need to care about supporting older standards either in Serenity, or in host builds.
And I agree with @bugaevc that the plain function approach looks a bit neater. The compiler will quickly let you know if something isn't gonna work in `static_assert` anyway :)